felix/routetable/route_table.go

```diff
@@ -1470,18 +1480,33 @@ func (r *RouteTable) applyUpdates(attempt int) error {
 
 	arpErrs := map[string]error{}
 	for ifaceName, addrToMAC := range r.pendingARPs {
-		// Add static ARP entries (for workload endpoints).  This may have been
-		// needed at one point but it no longer seems to be required.  Leaving
-		// it here for two reasons: (1) there may be an obscure scenario where
-		// it is needed. (2) we have tests that monitor netlink, and they break
-		// if it is removed because they see the ARP traffic.
+		if !r.ifacesToARP.Contains(ifaceName) {
+			continue
```

Comment:
if you add the ARPs to the pending when the devices need resync, than there should not be any need for `ifacesToARP` unless we do not want to dig out the ARPs from the routes every time and cache them in `ifacesToARP`. Modifying routes would need to modify this cache as well.